plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTableStatistics.java

```diff
@@ -0,0 +1,467 @@
+                "  orderdate AS \"case_quoted_lower\"," +
+                "  orderpriority AS \"CasE_QuoTeD_miXED\" " +
+                "FROM orders");
+        try {
+            gatherStats(tableName);
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('case_unquoted_upper', null, 15000, 0, null, null, null)," +
+                            "('case_unquoted_lower', null, 1000, 0, null, null, null)," +
+                            "('case_unquoted_mixed', 30000, 3, 0, null, null, null)," +
+                            "('case_quoted_upper', null, 14996, 0, null, null, null)," +
+                            "('case_quoted_lower', null, 2401, 0, null, null, null)," +
+                            "('case_quoted_mixed', 135000, 5, 0, null, null, null)," +
+                            "(null, null, null, null, 15000, null, null)");
+        }
+        finally {
+            executeInPostgres("DROP TABLE " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testNumericCornerCases()
+    {
+        try (TestTable table = fromColumns(
+                getQueryRunner()::execute,
+                "test_numeric_corner_cases_",
+                ImmutableMap.<String, List<String>>builder()
+                        .put("only_negative_infinity double", List.of("-infinity()", "-infinity()", "-infinity()", "-infinity()"))
+                        .put("only_positive_infinity double", List.of("infinity()", "infinity()", "infinity()", "infinity()"))
+                        .put("mixed_infinities double", List.of("-infinity()", "infinity()", "-infinity()", "infinity()"))
+                        .put("mixed_infinities_and_numbers double", List.of("-infinity()", "infinity()", "-5.0", "7.0"))
+                        .put("nans_only double", List.of("nan()", "nan()"))
+                        .put("nans_and_numbers double", List.of("nan()", "nan()", "-5.0", "7.0"))
+                        .put("large_doubles double", List.of("CAST(-50371909150609548946090.0 AS DOUBLE)", "CAST(50371909150609548946090.0 AS DOUBLE)")) // 2^77 DIV 3
+                        .put("short_decimals_big_fraction decimal(16,15)", List.of("-1.234567890123456", "1.234567890123456"))
+                        .put("short_decimals_big_integral decimal(16,1)", List.of("-123456789012345.6", "123456789012345.6"))
+                        .put("long_decimals_big_fraction decimal(38,37)", List.of("-1.2345678901234567890123456789012345678", "1.2345678901234567890123456789012345678"))
+                        .put("long_decimals_middle decimal(38,16)", List.of("-1234567890123456.7890123456789012345678", "1234567890123456.7890123456789012345678"))
+                        .put("long_decimals_big_integral decimal(38,1)", List.of("-1234567890123456789012345678901234567.8", "1234567890123456789012345678901234567.8"))
+                        .buildOrThrow(),
```

Comment:
Maybe we can have this as a "filtering" test similar to `filterDataMappingSmokeTestData` to remove the test cases not applicable to a specific connector but the list can be defined in base class.

Reply:
unless you feel strongly about this, i'd answer same as in https://github.com/trinodb/trino/pull/11635#discussion_r833984106 -- let's follow-up if needed.
this is because there is more code like that that Starburst would want to contribute, and doing refactor as part of the process would mean that Trino end users will get the features' benefits much later.

Reply:
thanks for understanding